### DISCUSSION TITLE
Filtering ComboBox (useSearchAutoComplete)

### DISCUSSION THREAD

**Discussion by lienas** [adobe/react-spectrum#3715] [Q&A] [answered]
I have a Autocomplete Box based on the following example:
https://reactspectrum.blob.core.windows.net/reactspectrum/26f9662ffb90217c26f906397b1a67b3197e476d/docs/react-aria/useSearchAutocomplete.html#example

Passing static items, like in the example, everything works fine. When I pass dynamic items, the filter does not work.
example here: 
https://codesandbox.io/s/agitated-swirles-s3z3ob?file=/src/App.tsx

**Comment by devongovett** [comment, marked as answer]
You should use the `defaultItems` prop rather than `items`. Using `items` makes the combo box controlled, so you would need to implement the filtering yourself. See https://reactspectrum.blob.core.windows.net/reactspectrum/26f9662ffb90217c26f906397b1a67b3197e476d/docs/react-aria/useSearchAutocomplete.html#dynamic-collections

**Reply by lienas** [reply]
makes sense 😀
